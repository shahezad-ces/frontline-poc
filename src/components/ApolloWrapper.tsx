"use client";
import { ApolloProvider } from "@apollo/client/react";
import client from "@frontline/libs/apolloClient";

export default function ApolloWrapper({ children }: { children: React.ReactNode }) {
  return <ApolloProvider client={client}>{children}</ApolloProvider>;
}