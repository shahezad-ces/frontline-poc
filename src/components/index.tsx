import Categories from "./Categories/Categories";
import PaginationWrapper from "./PaginationWrapper/PaginationWrapper";
import ProductCard from "./ProductCard/ProductCard";
import ProductsFilters from "./ProductsFilters/ProductsFilters";
import AddToCartButton from "./AddToCartButton/AddToCartButton";
import Header from "./Header/Header";

export {
  Categories,
  PaginationWrapper,
  ProductCard,
  ProductsFilters,
  AddToCartButton,
  Header,
};
