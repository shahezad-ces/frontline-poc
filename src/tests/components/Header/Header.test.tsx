import { screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { Header } from "@frontline/components";
import { renderWithProviders } from "@frontline/tests/RenderWithProviders";

jest.mock("@frontline/context/CartContext", () => {
  const originalModule = jest.requireActual("@frontline/context/CartContext");
  return {
    ...originalModule,
    useCart: jest.fn(),
  };
});

const mockUseCart = require("@frontline/context/CartContext").useCart;

describe("Header Component", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("renders brand name and cart icon", () => {
    mockUseCart.mockReturnValue({ totalItems: 0 });
    renderWithProviders(<Header />);
    expect(screen.getByText(/FrontLine/i)).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /FrontLine/i })).toHaveAttribute(
      "href",
      "/"
    );
  });

  it("does not show cart count when totalItems = 0", () => {
    mockUseCart.mockReturnValue({ totalItems: 0 });
    renderWithProviders(<Header />);
    expect(screen.queryByTestId("cart-count")).not.toBeInTheDocument();
  });

  it("shows cart count when totalItems > 0", () => {
    mockUseCart.mockReturnValue({ totalItems: 5 });
    renderWithProviders(<Header />);
    const badge = screen.getByTestId("cart-count");
    expect(badge).toBeInTheDocument();
    expect(badge).toHaveTextContent("5");
  });

  it("matches snapshot", () => {
    mockUseCart.mockReturnValue({ totalItems: 2 });
    const { asFragment } = renderWithProviders(<Header />);
    expect(asFragment()).toMatchSnapshot();
  });
});
