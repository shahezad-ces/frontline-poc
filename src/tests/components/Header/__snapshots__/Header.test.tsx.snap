// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Header Component matches snapshot 1`] = `
<DocumentFragment>
  <header
    class="bg-blue-800 text-white pl-6 shadow flex items-center justify-between"
  >
    <div
      class="flex items-center gap-6"
    >
      <div
        class="flex items-center gap-2"
      >
        <a
          href="/"
        >
          FrontLine
        </a>
      </div>
    </div>
    <div
      class="p-4 py-4 bg-yellow-400 relative"
    >
      <a
        href="/cart"
      >
        <svg
          aria-hidden="true"
          class="w-10 h-10"
          data-slot="icon"
          fill="currentColor"
          viewBox="0 0 20 20"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1 1.75A.75.75 0 0 1 1.75 1h1.628a1.75 1.75 0 0 1 1.734 1.51L5.18 3a65.25 65.25 0 0 1 13.36 1.412.75.75 0 0 1 .58.875 48.645 48.645 0 0 1-1.618 6.2.75.75 0 0 1-.712.513H6a2.503 2.503 0 0 0-2.292 1.5H17.25a.75.75 0 0 1 0 1.5H2.76a.75.75 0 0 1-.748-.807 4.002 4.002 0 0 1 2.716-3.486L3.626 2.716a.25.25 0 0 0-.248-.216H1.75A.75.75 0 0 1 1 1.75ZM6 17.5a1.5 1.5 0 1 1-3 0 1.5 1.5 0 0 1 3 0ZM15.5 19a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z"
          />
        </svg>
        <span
          class="absolute top-4 right-4 bg-red-600 text-white text-xs font-bold rounded-full w-5 h-5 flex items-center justify-center"
          data-testid="cart-count"
        >
          2
        </span>
      </a>
    </div>
  </header>
</DocumentFragment>
`;
